crates/chain/src/local_chain.rs

```diff
@@ -39,6 +40,39 @@ impl CheckPoint {
         Self(Arc::new(CPInner { block, prev: None }))
     }
 
+    /// Construct a checkpoint from the given `header` and block `height`.
+    ///
+    /// If `header` is of the genesis block, the checkpoint won't have a [`prev`] node. Otherwise,
+    /// we return a checkpoint linked with the previous block.
+    ///
+    /// [`prev`]: CheckPoint::prev
+    pub fn from_header(header: &bitcoin::block::Header, height: u32) -> Self {
+        let hash = header.block_hash();
+        let this_block_id = BlockId { height, hash };
+
+        if header.prev_blockhash == BlockHash::all_zeros() {
```

Comment:
isn't `if height == 0` strictly more correct here? This avoids the underflow on line 57 also. `match height.checked_sub(1)` might be the way to go.